ros/src/waypoint_updater/waypoint_updater.py

```diff
@@ -33,20 +35,67 @@ def __init__(self):
 
         # TODO: Add a subscriber for /traffic_waypoint and /obstacle_waypoint below
 
-
         self.final_waypoints_pub = rospy.Publisher('final_waypoints', Lane, queue_size=1)
 
         # TODO: Add other member variables you need below
+        self.base_waypoints = None
+        self.last_ts = None
 
         rospy.spin()
 
     def pose_cb(self, msg):
-        # TODO: Implement
-        pass
+        # stay still until base_waypoints is defined
+        if self.base_waypoints is None:
+            return
+        rospy.loginfo(rospy.get_caller_id() + ": current pose received")
+        # debug
+        print("pose_cb message: ", msg)
```

Comment:
I think it's better not to use `print` function in this context. 
You should use rospy log methods instead because it can make the prints clearer when we see them on screen

Reply:
Use the `rospy.loginfo` so the output is logged as well in the log files as @fadida mentioned before.